File: 프로그래머스/1/42748. K번째수/K번째수.cs
using System;

public class Solution {
    public int[] solution(int[] array, int[,] commands) {
        int n = commands.GetLength(0);
        int[] answer = new int[n];
        
        for(int t = 0; t < n; t++)
        {
            int i = commands[t, 0];
            int j = commands[t, 1];
            int k = commands[t, 2];
            int len = j - i + 1;
            int[] arr = new int[len];
            
            for(int x = 0; x < len; x++)
            {
                arr[x] = array[i - 1];
                i++;
            }
            Array.Sort(arr);
            answer[t] = arr[k - 1];
        }
        return answer;
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    for(int i = 0; i < commands.size(); i++)
    {
        vector<int> temp;
        for(int t = commands[i][0]; t <= commands[i][1]; t++)
        {
            temp.push_back(array[t - 1]);
        }
        sort(temp.begin(), temp.end());
        answer.push_back(temp[commands[i][2] - 1]);
    }
    return answer;
}